frontend/src/pages/RegistrationPage.jsx

```diff
@@ -1,30 +1,48 @@
-import { useContext, useState } from "react";
-import { useNavigate } from "react-router-dom";
-import { AuthContext } from "../contexts/AuthContext";
-import Cookies from "js-cookie";
+import { useState } from "react";
+import { useNavigate, Link } from "react-router-dom";
 import { register } from "../api/auth";
+import css from "./LoginAndRegisterPage.module.css";
 
 export default function RegistrationPage() {
   const navigate = useNavigate();
   const [username, setUsername] = useState("");
   const [password, setPassword] = useState("");
-  const { checkAuth } = useContext(AuthContext);
+  const [matchPassword, setMatchPassword] = useState("");
+  const [isMatched, setIsMatched] = useState(true);
```

Comment:
You don't need the isMatched state. You can simply create `const isMatched = password === matchPassword`, and use is where appropriate